mlir/lib/Gradient/Transforms/GradMethods/Adjoint.cpp

```diff
@@ -97,6 +97,12 @@ func::FuncOp AdjointLowering::discardAndReturnReg(PatternRewriter &rewriter, Loc
         rewriter.eraseOp(localDealloc);
     }
 
+    unallocFn.walk([&](Operation *op) {
+        if (isa<quantum::DeviceReleaseOp>(op)) {
+            rewriter.eraseOp(op);
+        }
+    });
```

Comment:
Is this re-added somewhere else? I am not yet intimately familiar with gradients so maybe @dime10 can chime in here. The concrete question is, we are not "leaking devices" by having this removing DeviceReleaseOp here, or are we?

Reply:
Nope, this is good. Never mind!